sktime/forecasting/trend/_spline_trend_forecaster.py

```diff
@@ -0,0 +1,291 @@
+        Whether to compute prediction intervals. If True, additional
+        calculations are done during fit to enable prediction intervals
+        to be calculated during predict. The prediction intervals are
+        based on an OLS regression model fitted to the data and calculated
+        according to Section 7.9 in [1]. Formulas are modified appropriately
+        if `with_intercept` is False.
+    n_knots : int, default=5
+        Number of knots of the splines if `knots` equals one of {'uniform', 'quantile'}.
+        Must be at least 2. Ignored if `knots` is array-like.
+    knots : {'uniform', 'quantile'}or array-like of shape (n_knots, n_features),
+        default='uniform'
+        Determines knot positions such that first knot <= features <= last knot.
+        - 'uniform': `n_knots` are distributed uniformly between the
+        min and max values of the features.
+        - 'quantile': `n_knots` are distributed uniformly along the quantiles
+        of the features.
+        - array-like: Specifies sorted knot positions, including the boundary knots.
+        Internally, additional knots are added before the first knot and after
+        the last knot based on the spline degree.
+    extrapolation : {'error', 'constant', 'linear', 'continue', 'periodic'},
+        default='constant'
+        Determines how to handle values outside the min and max values of the
+        training features:
+        - 'error': Raises a ValueError.
+        - 'constant': Uses the spline value at the minimum or maximum feature as
+        constant extrapolation.
+        - 'linear': Applies linear extrapolation.
+        - 'continue': Extrapolates as is (equivalent to `extrapolate=True` in
+        `scipy.interpolate.BSpline`).
+        - 'periodic': Uses periodic splines with a periodicity equal to the distance
+        between the first and last knot, enforcing equal function values and
+        derivatives at these knots.
+    include_bias : bool, default=True
+        If False, the last spline element inside the feature range is dropped.
+        B-splines sum to one over the basis functions, implicitly including
+        a bias term, i.e., a column of ones.
+
+    References
+    ----------
+    .. [1] Hyndman, Rob J., and George Athanasopoulos. Forecasting: principles
+    and practice, 3rd edition. OTexts: Melbourne, Australia. OTexts.com/fpp3.
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.trend import SplineTrendForecaster
+    >>> y = load_airline()
+    >>> forecaster = SplineTrendForecaster(
+    ...     degree=1,
+    ...     n_knots=5,
+    ...     knots="uniform",
+    ...     extrapolation="constant"
+    ... )
+    >>> forecaster.fit(y)
+    SplineTrendForecaster()
+    >>> y_pred = forecaster.predict(fh=[1, 2, 3])
+    """
+
+    _tags = {
+        "authors": ["jgyasu", "tensorflow-as-tf", "mloning", "aiwalter", "fkiraly"],
```

Comment:
I suggest to make it just you. The dependency has been acknowledged through classs inheritance IMO, @fkiraly can comment further.

```suggestion
        "authors": ["jgyasu"],
```

However, you should consider adding a maintainer tag too.